src/nat/authentication/oauth2/oauth2_auth_code_flow_provider.py

```
@@ -41,26 +45,30 @@ async def _attempt_token_refresh(self, user_id: str, auth_result: AuthResult) ->
+                credentials=[BearerTokenCred(token=SecretStr(new_token_data["access_token"]))],
+                token_expires_at=new_expires_at,
+                raw=new_token_data,
+            )
 
-        self._authenticated_tokens[user_id] = new_auth_result
+            self._authenticated_tokens[user_id] = new_auth_result
+        except httpx.HTTPStatusError:
+            return None
+        except httpx.RequestError:
+            return None
+        except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)